ecommerce/views.py

```diff
@@ -108,8 +103,10 @@ def _get_request_properties(self):
         return False, product_type
 
     def get_queryset(self):
+        now = now_in_utc()
+        expired_courseruns = CourseRun.objects.filter(enrollment_end__lt=now).values_list("id", flat=True)
         queryset = (
-            Product.objects.exclude(productversions=None)
+            Product.objects.exclude(Q(productversions=None) | (Q(object_id__in=expired_courseruns) & Q(content_type__model="courserun")))
             .select_related("content_type")
```

Comment:
This got rid of `.prefetch_related("content_object")` which would impact performance, please add that back in again.